File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/uncomment/direct/Java/Go/s527301188_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')
	strs := strings.Split(strings.TrimSpace(input), " ")
	nums := make([]int, len(strs))
	top := 0

	for _, str := range strs {
		switch str[0] {
		case '+':
			top--
			nums[top-1] = nums[top-1] + nums[top]
		case '-':
			top--
			nums[top-1] = nums[top-1] - nums[top]
		case '*':
			top--
			nums[top-1] = nums[top-1] * nums[top]
		default:
			num, _ := strconv.Atoi(str)
			nums[top] = num
			top++
		}
	}
	fmt.Println(nums[0])
} // End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Python/s527301188_commented.py
import sys

def main():
    input_line = sys.stdin.readline().strip()
    strs = input_line.split(" ")
    nums = []
    for s in strs:
        if s == '+':
            nums[-2] += nums[-1]
            nums.pop()
        elif s == '-':
            nums[-2] -= nums[-1]
            nums.pop()
        elif s == '*':
            nums[-2] *= nums[-1]
            nums.pop()
        else:
            nums.append(int(s))
    print(nums[0])

if __name__ == "__main__":
    main()  # End of Code
